### DISCUSSION TITLE
Produce ref assemblies from command-line and msbuild

### DISCUSSION THREAD

**Discussion by 0x53A** [dotnet/fsharp#12907] [Ideas]
It would be nice if fsc could produce reference assemblies (see https://github.com/dotnet/roslyn/pull/17558) and participate in the corresponding msbuild feature: https://github.com/Microsoft/msbuild/issues/1986.

This can potentially reduce (re)build times by a magnitude.

Deterministic builds (https://github.com/Microsoft/visualfsharp/pull/2954) are probably a prerequisite for this.

Here is the Roslyn spec: https://github.com/dotnet/roslyn/blob/master/docs/features/refout.md

**Comment by abelbraaksma** [comment]
Some good discussion points are in the referenced issue, in particular this comment, which summarizes several views from other threads: https://github.com/dotnet/fsharp/issues/8642#issuecomment-731861186
